kod/object/active/holder/room/monsroom/g9.kod

```diff
@@ -256,317 +256,198 @@ messages:
-      }
-
-      Send(self,@SetSector,#sector=2,#animation=ANIMATE_FLOOR_LIFT,
-           #height=224,#speed=0);
-      Send(self,@SetSector,#sector=3,#animation=ANIMATE_FLOOR_LIFT,
-           #height=224,#speed=0);
-
-      return;
-   }
-
-   EnableIllusions()
-   {
-      local i, each_obj;
-
-      for i in plActive
-      {
-         each_obj = Send(self,@HolderExtractObject,#Data=i);
-         if IsClass(each_obj,&portal)
-         {
-            Send(each_obj,@Delete);
-         }
-      }
+      % Create platform levers
+      % Note: The platform levers are the current solution to the puzzle.
 
-      Send(self,@SetSector,#sector=2,#animation=ANIMATE_FLOOR_LIFT,
-           #height=400,#speed=0);
-      Send(self,@SetSector,#sector=3,#animation=ANIMATE_FLOOR_LIFT,
-           #height=264,#speed=0);
-
-      return;
-   }
+      poCenterLever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poCenterLever,#new_row=21,#new_col=55,#fine_row=30,#fine_col=30);
+      
+      poNWLever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poNWLever,#new_row=16,#new_col=51,#fine_row=0,#fine_col=0);
+      
+      poNELever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poNELever,#new_row=16,#new_col=60,#fine_row=0,#fine_col=0);
 
-   FirstUserEntered()
-   {
-      if NOT (Send(SYS,@AreRoomIllusionsEnabled))
-      {
-         Send(self,@SetSector,#sector=2,#animation=ANIMATE_FLOOR_LIFT,
-              #height=224,#speed=0);
-         Send(self,@SetSector,#sector=3,#animation=ANIMATE_FLOOR_LIFT,
-              #height=224,#speed=0);
-      }
+      poSWLever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poSWLever,#new_row=25,#new_col=51,#fine_row=0,#fine_col=0);
+      
+      poSELever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poSELever,#new_row=25,#new_col=60,#fine_row=0,#fine_col=0);
 
       propagate;
    }
-
-   NewHold(what=$, new_row=$, new_col=$)
+   
+   ReturnPuzzleToInitialState()
+   "Returns levers, lights, and spire to their original state."
    {
-      if NOT (Send(SYS,@AreRoomIllusionsEnabled))
-         OR ((what=$) OR (new_row=$) OR (new_col=$))
-         OR Send(poPlatform,@GetState) = SECTOR_UP
-      {
-         propagate;
-      }
+      ptResetPuzzle = $;
 
-      if isClass(what,&WallofLightning)
-      {
-         post(self,@UpdateSpireState,#oLightning=what);
-      }
-
-      propagate;
-   }
-
-   UpdateSpireState(oLightning=$)
-   {
-      local new_row, new_col;
+      % Flip the platform levers to their original state
+      Send(self,@ResetLeverStates);
 
-      new_row = Send(oLightning,@GetRow);
-      new_col = Send(oLightning,@GetCol);
+      % Set a timer to "unlock" use of the levers after a short delay
+      ptActivateLevers = CreateTimer(self,@ActivateLevers,PUZZLE_TIMELIMIT);
```

Comment:
Need to delete any existing timer?

Might be handy to have a function to cancel all active timers, as this is done from a few places.